### ISSUE TITLE
SQS Batch Publish Broken

### ISSUE THREAD

**Issue by GrahamCampbell** [async-aws/aws#1620]
On v2 using the AWS JSON-1.0 protocol: `Undefined array key "Failed"`.

**Comment by jderusse** [comment]
That's a bug in the AWS: The specs say "the field is required"

https://github.com/aws/aws-sdk-php/blob/1e01734856f8ec2b6d525d1c99c5a34029217598/src/data/sqs/2012-11-05/api-2.json#L1141-L1151

either:
- ask AWS to fix the spec (from my previous experience it'll take years)
- patch the specs makes (remove required section)
- change the generator to never trust the result's `required` section

**Comment by GrahamCampbell** [comment]
You are not wrong, and we will need to fix this on our side, even though this is not our bug.
